### PR TITLE
Update documentation and testing

### PR DESCRIPTION
**Pull request by L1uY1jun**

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -15,53 +15,42 @@ This is an overview of the application.
...
-TrackAScholar (TAS) helps university administrative staff manage scholarship applications.
-It stores all applicant's details such as contact info and scholarship applied etc. all in one place.
-TrackAScholar (TAS) is a one of a kind desktop app.It is the **only** application that any university administrative staff needs.
+TrackAScholar (TAS) is a one of a kind desktop app. It is the **only** application that any university administrative staff needs.
+TAS helps university administrative staff manage scholarship applications.
+It can store all the applicant's details such as contact info and scholarship applied etc. in one place.
 
-TAS have many functions that university administrative staff can use to help speed up the application process immensly.
+TAS has many features that university administrative staff can use to help speed up the application process immensely.
+TAS presents the administrative staff with the option to pin important applicants, edit their details easily and
```
Comment (benjytan45678) [inline docs/UserGuide.md]: with various options such as ... instead of the option to?

File: docs/UserGuide.md [hunk 2]
```diff
@@ -15,53 +15,42 @@ This is an overview of the application.
...
 
-It is stated that on average 1.7 million university scholarships are given out each year in the United States alone.Furthermore, 
-it is found that only 1 in 8 student actually wins a scholarship.This brings us to about more than 10 million applications a year.
-
-As such, many universities often take an extended period of many months to process these applications.
-University administrative staff often have to work overtime to process these large number of applications.
-Poor scholarship management procedures employed in many of these universities further complicates the process. 
-
-TrackAScholar (TAS) is an app designed for university administrative staff members to streamline the scholarship management procedure. 
+Based on our research, many universities often take an extended period of many months to process scholarship applications.
```
Comment (benjytan45678) [inline docs/UserGuide.md]: many months change to several/numerous months?

File: src/main/java/seedu/trackascholar/commons/util/StringUtil.java [hunk 3]
```diff
@@ -20,8 +20,9 @@ public class StringUtil {
      *       containsWordIgnoreCase("ABc def", "DEF") == true
      *       containsWordIgnoreCase("ABc def", "AB") == false //not a full word match
      *       </pre>
-     * @param sentence cannot be null
-     * @param word cannot be null, cannot be empty, must be a single word
+     *
+     * @param sentence cannot be null.
+     * @param word cannot be null, cannot be empty, must be a single word.
```
Comment (benjytan45678) [inline src/main/java/seedu/trackascholar/commons/util/StringUtil.java:25]: Maybe need @return to specify the boolean value returned?

File: src/main/java/seedu/trackascholar/logic/commands/FindCommand.java [hunk 4]
```diff
@@ -12,14 +12,14 @@
...
- * Finds and lists all applicants in TrackAScholar whose name contains any of the argument keywords.
+ * Finds and lists all applicants in TrackAScholar whose details contain any of the argument keywords.
  * Keyword matching is case-insensitive.
  */
 public class FindCommand extends Command {
 
     public static final String COMMAND_WORD = "find";
 
-    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Finds all applicants who contain any of the "
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Finds all applicants whose detail contain any of the "
```
Comment (benjytan45678) [inline src/main/java/seedu/trackascholar/logic/commands/FindCommand.java]: whose detail => whose details?

File: src/main/java/seedu/trackascholar/logic/parser/AddCommandParser.java [hunk 5]
```diff
@@ -22,14 +22,15 @@
...
+ * Parses input arguments and creates a new AddCommand object.
  */
 public class AddCommandParser implements Parser<AddCommand> {
 
     /**
      * Parses the given {@code String} of arguments in the context of the AddCommand
      * and returns an AddCommand object for execution.
-     * @throws ParseException if the user input does not conform the expected format
+     *
+     * @throws ParseException if the user input does not conform the expected format.
```
Comment (benjytan45678) [inline src/main/java/seedu/trackascholar/logic/parser/AddCommandParser.java]: does not conform ***to***?

File: src/main/java/seedu/trackascholar/logic/parser/DeleteCommandParser.java [hunk 6]
```diff
@@ -7,14 +7,15 @@
...
+ * Parses input arguments and creates a new DeleteCommand object.
  */
 public class DeleteCommandParser implements Parser<DeleteCommand> {
 
     /**
      * Parses the given {@code String} of arguments in the context of the DeleteCommand
      * and returns a DeleteCommand object for execution.
-     * @throws ParseException if the user input does not conform the expected format
+     *
+     * @throws ParseException if the user input does not conform the expected format.
```
Comment (benjytan45678) [inline src/main/java/seedu/trackascholar/logic/parser/DeleteCommandParser.java]: does not conform ***to***?

File: src/main/java/seedu/trackascholar/logic/parser/EditCommandParser.java [hunk 7]
```diff
@@ -20,14 +20,15 @@
...
+ * Parses input arguments and creates a new EditCommand object.
  */
 public class EditCommandParser implements Parser<EditCommand> {
 
     /**
      * Parses the given {@code String} of arguments in the context of the EditCommand
      * and returns an EditCommand object for execution.
-     * @throws ParseException if the user input does not conform the expected format
+     *
+     * @throws ParseException if the user input does not conform the expected format.
```
Comment (benjytan45678) [inline src/main/java/seedu/trackascholar/logic/parser/EditCommandParser.java]: does not conform ***to***?

File: src/main/java/seedu/trackascholar/logic/parser/FilterCommandParser.java [hunk 8]
```diff
@@ -8,14 +8,15 @@
...
 public class FilterCommandParser implements Parser<FilterCommand> {
 
     /**
-     * Parses the given {@code String} of arguments in the context of the FindCommand
-     * and returns a FindCommand object for execution.
-     * @throws ParseException if the user input does not conform the expected format
+     * Parses the given {@code String} of arguments in the context of the FilterCommand
+     * and returns a FilterCommand object for execution.
+     *
+     * @throws ParseException if the user input does not conform the expected format.
```
Comment (benjytan45678) [inline src/main/java/seedu/trackascholar/logic/parser/FilterCommandParser.java]: does not conform ***to***?

File: src/main/java/seedu/trackascholar/logic/parser/PinCommandParser.java [hunk 9]
```diff
@@ -9,13 +9,14 @@
...
- * Parses input arguments and creates a new PinCommand object
+ * Parses input arguments and creates a new PinCommand object.
  */
 public class PinCommandParser implements Parser<PinCommand> {
     /**
      * Parses the given {@code String} of arguments in the context of the PinCommand
      * and returns a PinCommand object for execution.
-     * @throws ParseException if the user input does not conform the expected format
+     *
+     * @throws ParseException if the user input does not conform the expected format.
```
Comment (benjytan45678) [inline src/main/java/seedu/trackascholar/logic/parser/PinCommandParser.java]: does not conform ***to***?

File: src/main/java/seedu/trackascholar/logic/parser/RemoveCommandParser.java [hunk 10]
```diff
@@ -7,14 +7,15 @@
...
+ * Parses input arguments and creates a new RemoveCommand object.
  */
 public class RemoveCommandParser implements Parser<RemoveCommand> {
 
     /**
      * Parses the given {@code String} of arguments in the context of the RemoveCommand
      * and returns a RemoveCommand object for execution.
-     * @throws ParseException if the user input does not conform the expected format
+     *
+     * @throws ParseException if the user input does not conform the expected format.
```
Comment (benjytan45678) [inline src/main/java/seedu/trackascholar/logic/parser/RemoveCommandParser.java]: conform ***to***?

File: src/main/java/seedu/trackascholar/logic/parser/SortCommandParser.java [hunk 11]
```diff
@@ -24,7 +24,8 @@ public class SortCommandParser implements Parser<SortCommand> {
     /**
      * Parses the given {@code String} of arguments in the context of the SortCommand
      * and returns a SortCommand object for execution.
-     * @throws ParseException if the user input does not conform the expected format
+     *
+     * @throws ParseException if the user input does not conform the expected format.
```
Comment (benjytan45678) [inline src/main/java/seedu/trackascholar/logic/parser/SortCommandParser.java]: conform ***to***?

File: src/main/java/seedu/trackascholar/logic/parser/TrackAScholarParser.java [hunk 12]
```diff
@@ -36,9 +36,9 @@ public class TrackAScholarParser {
     /**
      * Parses user input into command for execution.
      *
-     * @param userInput full user input string
-     * @return the command based on the user input
-     * @throws ParseException if the user input does not conform the expected format
+     * @param userInput full user input string.
+     * @return the command based on the user input.
+     * @throws ParseException if the user input does not conform the expected format.
```
Comment (benjytan45678) [inline src/main/java/seedu/trackascholar/logic/parser/TrackAScholarParser.java]: conform ***to***?

File: src/main/java/seedu/trackascholar/ui/AlertWindow.java [hunk 13]
```diff
@@ -7,14 +7,15 @@
...
  */
 public class AlertWindow {
     private boolean isConfirmed = false;
 
     /**
-     * Returns official confirmation user reply and displays an alert window
-     * @param  message prompted to user
+     * Returns official confirmation user reply and displays an alert window.
+     *
+     * @param  message prompted to user.
```
Comment (benjytan45678) [inline src/main/java/seedu/trackascholar/ui/AlertWindow.java:18]: Missing @returns here too?

File: src/main/java/seedu/trackascholar/commons/util/StringUtil.java (hunk 3 above)
Comment (L1uY1jun) [reply src/main/java/seedu/trackascholar/commons/util/StringUtil.java:25]: The return value was specified in the above javadocs for that method

**Comment by benjytan45678** [comment]
LGTM, merged

**Pull request merged**